bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -258,9 +280,24 @@ public ChannelPool(ChannelFactory factory, int count) throws IOException {
   private ManagedChannel getNextChannel() {
     int currentRequestNum = requestCount.getAndIncrement();
     int index = Math.abs(currentRequestNum % channels.size());
+
+    int minLoad = channels.get(index).getOpenStreamCount();
+
+    if (minLoad > repickThreshold) {
+      for (int i = 0; i < repickThreshold; i++) {
```

Comment:
I think this should use `repickChances` instead of `repickThreshold`

Reply:
yep, sorry, this was meant to be a sketch. I will update it later on this week.
I would also like to benchmark it against the current round robin scheme and a sticky version of this approach: reuse the last channel if there are no outstanding requests.